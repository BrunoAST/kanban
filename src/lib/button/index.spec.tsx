import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button, { ButtonProps } from '.';

function renderButton({ size, variant }: Partial<ButtonProps>) {
  render(<Button size={size} variant={variant} onClick={() => console.log('Click')}>Content</Button>);
}

describe('<Button />', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<Button onClick={jest.fn}>Content</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render children content', () => {
    renderButton({});
    expect(screen.getByText('Content')).toBeDefined();
  });

  describe('Size', () => {
    test('Should set the large size by default', () => {
      renderButton({});
      const button = screen.getByText('Content');
      expect(button.classList).toContain('py-3.5');
      expect(button.classList).not.toContain('py-2');
    });

    test('Should set the small size when passed', () => {
      renderButton({ size: 'small' });
      const button = screen.getByText('Content');
      expect(button.classList).toContain('py-2');
      expect(button.classList).not.toContain('py-3.5');
    });
  });

  describe('Variant', () => {
    test('Should set the primary variant by default', () => {
      renderButton({});
      const button = screen.getByText('Content');
      expect(button.classList).toContain('bg-purple-200');
      expect(button.classList).not.toContain('bg-purple-125');
      expect(button.classList).not.toContain('bg-red-200');
    });

    test('Should set the secondary variant when passed', () => {
      renderButton({ variant: 'secondary' });
      const button = screen.getByText('Content');
      expect(button.classList).toContain('bg-purple-125');
      expect(button.classList).not.toContain('bg-purple-200');
      expect(button.classList).not.toContain('bg-red-200');
    });

    test('Should set the destructive variant when passed', () => {
      renderButton({ variant: 'destructive' });
      const button = screen.getByText('Content');
      expect(button.classList).toContain('bg-red-200');
      expect(button.classList).not.toContain('bg-purple-125');
      expect(button.classList).not.toContain('bg-purple-200');
    });
  });
});
