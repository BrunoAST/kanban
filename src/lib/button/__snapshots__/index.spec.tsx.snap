// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="
        bg-purple-200 text-white hover:bg-purple-100 py-3.5 px-4 rounded-3xl duration-300 font-bold text-sm"
  onClick={[Function]}
  type="button"
>
  Content
</button>
`;
